text/0025-core-compute-resources-as-plugins.md

```diff
@@ -0,0 +1,77 @@
+- Feature Name: Core Compute Resources as Plugins
+- Start Date: 2016-10-21
+- RFC PR: (leave this empty)
+- Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Core Compute Resources should move to plugins
+
+# Motivation
+[motivation]: #motivation
+
+Core has been set up to allow compute resources as plugins for some while, and
+we already have several examples of this in use (Xen, Azure, DigitalOcean,
+etc). This has brought benefits:
+
+* able to release independatly of Core
+* make use of the modular Fog gems that are now available
+* separate logging of issues
+* dedicated maintainers who know the compute resource well
```

Comment:
If no-one is maintaining the code, then it'll be of poor quality regardless of whether it's in core or in a plugin. We know that's true today for some of our CRs.

We've seen historically that people are more willing to get involved if the codebase is not big & scary (regardless of whether the *actual* code is any less complex). On that basis we should get more contributors to a CR plugin than to the same code in core. It's all perception :)